packages/engine/lib/stateful/src/message/outbound.rs

```diff
@@ -38,27 +39,31 @@ impl fmt::Display for Error {
 
 impl StdError for Error {}
 
-/*
- * We want Serde to deserialize a message to the correct enum variant,
- * and to do so, we need to emulate tagged unions. The matter is complicated
- * by the fact that our tag, the field "type", can be any string! So we want
- * to pick the Message::CreateAgent variant if "type" is "create_agent",
- * Message::RemoveAgent if "type" is "remove_agent", and Message::Generic if
- * "type" has any other value.
- *
- * To do this, we create two enums with a single variant, CreateAgent and
- * RemoveAgent. Message::CreateAgent has a field "type" of type CreateAgent:
- * this means that serde will match it correctly when "type" is "create_agent"!
- * Same holds for "remove_agent". Finally, if it cannot match either variant,
- * Serde will fall back to Message::Generic.
- *
- * Since we are at it, I've also done the same with the "hash" id, as it is
- * another requirement for Message::CreateAgent and Message::RemoveAgent.
- */
+// We want Serde to deserialize a message to the correct enum variant, and to do so, we need to
+// emulate tagged unions. The matter is complicated by the fact that our tag, the field "type", can
+// be any string! So we want to pick the Message::CreateAgent variant if "type" is "create_agent",
+// Message::RemoveAgent if "type" is "remove_agent", and Message::Generic if "type" has any other
+// value.
+//
+// To do this, we create two enums with a single variant, CreateAgent and RemoveAgent.
+// Message::CreateAgent has a field "type" of type CreateAgent: this means that serde will match it
+// correctly when "type" is "create_agent"! Same holds for "remove_agent". Finally, if it cannot
+// match either variant, Serde will fall back to Message::Generic.
+//
+// Since we are at it, I've also done the same with the "hash" id, as it is another requirement for
+// Message::CreateAgent and Message::RemoveAgent.
+/// A message sent by an [`Agent`].
+///
+/// Currently, three types of [built-in messages] are available: `"create_agent"`, `"remove_agent"`,
+/// and `"stop"`. For [messages sent] to other agents, a [`Generic`] message is used.
+///
+/// [built-in messages]: https://hash.ai/docs/simulation/creating-simulations/agent-messages/built-in-message-handlers
+/// [messages sent]: https://hash.ai/docs/simulation/creating-simulations/agent-messages/sending-messages
+/// [`Generic`]: Self::Generic
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
 #[serde(untagged)]
 #[allow(clippy::large_enum_variant)]
-pub enum Outbound {
+pub enum Message {
```

Comment:
As mentioned above, I renamed this because we don't directly use `Inbound`, it's encoded in a `FieldSpec`. Anyway, `Inbound` still has the `to` fields of `Outbound` encoded, so an `Inbound` message still contains the full `Outbound` (list), thus I think it's confusing to rename it `OutboundMessage` because then `Inbound` would contain an `OutboundMessage` (even though it's not directly referenced).

Reply:
Very strange